test/inductor/test_compiled_optimizers.py

```diff
@@ -131,13 +131,13 @@ def compile_opt(opt_compiled, closure=None):
     # run the patcher so that step has the expected structure
     torch._dynamo.eval_frame.TorchPatcher.patch()
 
-    # unwrap step to avoid a deliberate graph break due to
+    # unwrap step TWICE to avoid a deliberate graph break due to
     # a limitation of functionalization/no_grad detection
     # see the [Note on graph break] in optimizer.py
     # This ignores the outer _use_grad_if_differentiable wrapper
     # and instead manually disables grad before calling step, which is fine
     # for now as dynamo does not support differentiable optimizers anyway
-    step_fn = opt_compiled.step.__wrapped__
+    step_fn = opt_compiled.step.__wrapped__.__wrapped__
```

Comment:
why are we unwrapping twice? shouldn't a single unwrap be okay since the profile var is okay now. (should just produce a warning)

Reply:
ah wait, because the profile wrapper is the outermost wrapper, right?

Reply:
yes